Don't make black transparent when drawing on wxOverlay under MSW

vadz opened the pull request:
Using black as the transparent colour is not a great idea as everything drawn using it on the overlay just doesn't appear at all then.

Use a different colour that will hopefully be much less common.

---

@AliKet please let me know if you have any comments.

vadz committed:
Don't make black transparent when drawing on wxOverlay under MSW

Using black as the transparent colour is not a great idea as everything
drawn using it on the overlay just doesn't appear at all then.

Use a different colour that will hopefully be much less common.

AliKet commented:
Great, this is definitely the right thing to do and it should have been done this way from the beginning. 👍

AliKet commented:
While you're at it, these can be updated after merging this PR:
https://github.com/wxWidgets/wxWidgets/blob/ee309e078a2294d59c754b79c0407184e7558738/src/aui/framemanager.cpp#L92-L101 and this https://github.com/wxWidgets/wxWidgets/blob/ee309e078a2294d59c754b79c0407184e7558738/src/generic/splitter.cpp#L72-L81

i.e.:
```
static wxBitmap wxPaneCreateStippleBitmap()
{
    unsigned char data[] = { 0,0,0,192,192,192, 192,192,192,0,0,0 };
    wxImage img(2,2,data,true);
    return wxBitmap(img);
}
```

vadz committed:
Factor out wxDrawOverlayResizeHint() from the existing code

Avoid duplicating the same code in wxSplitterWindow and wxAuiManager by
extracting it into a (private) header included by both of them.

No real changes.

vadz committed:
Use black in the stipple brush bitmap used for resize hint

After the changes of 315e1074d3 (Don't make black transparent when
drawing on wxOverlay under MSW, 2024-04-19) we don't need to use the
slightly-not-black colour in this brush any longer.
